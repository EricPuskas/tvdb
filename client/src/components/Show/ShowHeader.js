import React from "react";
import Rating from "react-rating";

const ShowHeader = ({ title, genres, rating, ratingText }) => {
  return (
    <div className="Show-title">
      <h1> {title} </h1>
      <ul className="tags">{genres}</ul>
      <div>
        <Rating
          emptySymbol="far fa-star"
          fullSymbol="fas fa-star"
          stop={10}
          initialRating={rating}
          readonly
        />{" "}
        <span style={{ display: "block" }}>{ratingText}</span>
      </div>
    </div>
  );
};

export default ShowHeader;
