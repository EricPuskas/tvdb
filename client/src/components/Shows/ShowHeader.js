import React from "react";
import Moment from "react-moment";
import isValidDate from "../../utils/isValidDate";

const ShowHeader = ({ title, firstAired, network }) => {
  return (
    <>
      <h2> {title} </h2>
      <hr />
      <div>
        First Aired:{" "}
        <strong>
          {isValidDate(firstAired) ? (
            <Moment format="MMM DD (YYYY)">{firstAired}</Moment>
          ) : (
            "Unknown"
          )}
        </strong>
      </div>
      <div>
        Network: <strong>{network ? network : "Unknown"}</strong>
      </div>
      <hr />
    </>
  );
};

export default ShowHeader;
