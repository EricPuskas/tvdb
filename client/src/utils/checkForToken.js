import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
