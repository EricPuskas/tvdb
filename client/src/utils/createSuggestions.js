const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
