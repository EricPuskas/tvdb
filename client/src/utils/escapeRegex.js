const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
