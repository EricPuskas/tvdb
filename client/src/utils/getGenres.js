import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
