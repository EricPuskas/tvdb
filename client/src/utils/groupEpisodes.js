const groupEpisodes = data => {
  let seasons = [];
  for (let i = 0; i < data.length; i++) {
    if (!seasons[data[i].airedSeason]) seasons[data[i].airedSeason] = [];
    seasons[data[i].airedSeason].push({
      id: data[i].id,
      season: data[i].airedSeason,
      episode: data[i].airedEpisodeNumber,
      title: data[i].episodeName
    });
  }
  return seasons;
};

export default groupEpisodes;
