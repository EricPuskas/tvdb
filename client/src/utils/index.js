import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
