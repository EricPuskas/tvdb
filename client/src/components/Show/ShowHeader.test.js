import React from "react";
import { shallow } from "enzyme";
import { findByAttribute } from "../../utils";
import ShowHeader from "./ShowHeader";

describe("Show Header Component", () => {
  describe("Renders with props", () => {
    const props = {
      title: "Arrow",
      genres: (
        <div>
          <li key={jest.fn()}>
            <span className="tag">test</span>
          </li>
        </div>
      ),
      rating: 8.3,
      ratingText: "Lorem Ipsum"
    };
    const component = shallow(<ShowHeader {...props} />);
    const wrapper = findByAttribute(component, "showheader");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });
  });
});
