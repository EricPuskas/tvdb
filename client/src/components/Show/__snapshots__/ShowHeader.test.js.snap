// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Header Component Renders with props Should pass snapshot test 1`] = `"<div class=\\"ShowHeader\\" data-test=\\"showheader\\"><h1> Arrow </h1><ul class=\\"tags\\"><div><li><span class=\\"tag\\">test</span></li></div></ul><div><span style=\\"display:inline-block;direction:ltr\\"><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:100%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:30.00000000000007%\\"><span class=\\"fas fa-star\\"></span></span></span><span style=\\"cursor:inherit;display:inline-block;position:relative\\"><span class=\\"far fa-star\\"></span><span style=\\"display:inline-block;position:absolute;overflow:hidden;top:0;left:0;width:0%\\"><span class=\\"fas fa-star\\"></span></span></span></span> <span style=\\"display:block\\">Lorem Ipsum</span></div></div>"`;
